### DISCUSSION TITLE
Ensure page is rendered on the server side always

### DISCUSSION THREAD

**Discussion by alimozdemir** [nuxt/nuxt#22621] [Questions] [answered]
It's been a while I'm using Nuxt 3 successfully. I would like to discuss one topic regarding the lifecycle/rendering mode of the pages.

Nuxt SSR works only on initial rendering.  I would like to render some pages only on the server side, I don't want to expose the API calls to the client. Is there any plan to add that feature in the near future? And if it is so, how can I contribute to it?

PS; I can achieve the functionality of replacing nuxt-links with an anchor but I don't want to do that due to reloading the whole app.

**Comment by TheAlexLichter** [comment, marked as answer]
> It's been a while I'm using Nuxt 3 successfully.

Happy to hear that 👏🏻
 
> Nuxt SSR works only on initial rendering. I would like to render some pages only on the server side, I don't want to expose the API calls to the client. Is there any plan to add that feature in the near future? And if it is so, how can I contribute to it?

If the API calls do not have to be "real time", you might be able to prerender the site. This way, the API calls are stored in a payload and are detached from your actual API.

Other than that, you might want to check out server components! These are still experimental but allow you to render arbitrary content on the server. See [@danielroe's blogpost](https://roe.dev/blog/nuxt-server-components) for more info

**Reply by alimozdemir** [reply]
Thank you for the answer, I think server components will solve my problem. Thanks a lot!
